src/llmcompressor/modifiers/quantization/gptq/utils/gptq_wrapper.py

```diff
@@ -122,20 +127,37 @@ def compress(
 
         tick = time.time()
 
-        # consider activation ordering
-        if weight_quant_args.actorder:
-            # use hessian to create a permutation of weights
-            perm = torch.argsort(torch.diag(self.H), descending=True)
+        if strategy == QuantizationStrategy.GROUP:
+            # mapping from column index to group index
+            g_idx = (
+                torch.arange(self.columns, device=W.device, dtype=torch.int)
+                // weight_quant_args.group_size
+            )
+
+            if actorder == ActivationOrderingStrategy.GROUP:
+                # permute by activation order first, then update groups
+                # python doesn't support inlining, so repeat code to save mem
+                perm = torch.argsort(torch.diag(self.H), descending=True)
+                W = W[:, perm]
+                self.H = self.H[perm][:, perm]
+
+                # compute quantization parameters
+                self._update_quantization_parameters(weight_quant_args, W)
 
-            # permute weight and hessian
-            W = W[:, perm]
-            self.H = self.H[perm][:, perm]
+                # use identity g_idx (invert permutation later)
 
-            # update quantization parameters for activation ordering
-            observer = MemorylessObserver(weight_quant_args)
-            _scale, _zero_point = observer(W)
-            update_parameter_data(self.layer, _scale, "weight_scale")
-            update_parameter_data(self.layer, _zero_point, "weight_zero_point")
+            elif actorder == ActivationOrderingStrategy.WEIGHT:
+                # update groups first, then permute by activation order
+                self._update_quantization_parameters(weight_quant_args, W)
+
+                # use hessian to create a permutation of weights
+                # python doesn't support inlining, so repeat code to save mem
+                perm = torch.argsort(torch.diag(self.H), descending=True)
+                W = W[:, perm]
+                self.H = self.H[perm][:, perm]
```

Comment:
nit: if we're repeating this in both if statement blocks can we break it out into a helper fn?